bug: corner points seem to be calculated wrong on android

### Plugin(s)

- [x] Barcode Scanning
- [ ] Face Detection
- [ ] Face Mesh Detection
- [ ] Selfie Segmentation
- [ ] Translation

### Version

7.0.0, 7.1.0

### Platform(s)

- [x] Android
- [ ] iOS
- [ ] Web

### Current behavior

when trying to scan a barcode within the sight (box) with the demo app it will not recognize that the barcode is within the boundaries of the sight. On IOS this works fine.
When moving further away from the barcode the app will scan the barcode.

### Expected behavior

The scanner should scan a barcode when its within the boundaries of the sight.

### Reproduction

https://github.com/robingenz/capacitor-mlkit-plugin-demo

### Steps to reproduce

1. clone and start the demo app on android.
2. upgrade dependencies (optional as it does not work on 7 and 7.1 for me)
3. click start scan (the first button)
4. try to scan a barcode while being "close" to the barcode
5. Move further away from the barcode -> the scan will work

### Other information

i am currently in the process of upgrading our app from capacitor 6 to 7 and noticed that the scanner does not work well on android. When using version 6.2 (@capacitor-mlkit/barcode-scanning) then the scan works fine and is very fast with version 7 it starts to have the issue described above.

Today i tried if the issue exists with the demo app that is available and it has the same issue.

The device i am using is an Fairphone 4 with Android 13

### Capacitor doctor

💊   Capacitor Doctor  💊

Latest Dependencies:

  @capacitor/cli: 7.0.1
  @capacitor/core: 7.0.1
  @capacitor/android: 7.0.1
  @capacitor/ios: 7.0.1

Installed Dependencies:

  @capacitor/cli: 7.0.0
  @capacitor/core: 7.0.1
  @capacitor/ios: 7.0.1
  @capacitor/android: 7.0.1

[success] iOS looking great! 👌
[success] Android looking great! 👌

### Before submitting

- [x] I have read and followed the [bug report guidelines](https://capawesome.io/contributing/bug-reports/).
- [x] I have attached links to possibly related issues and discussions.
- [x] I understand that incomplete issues (e.g. without reproduction) are closed.

[capacitor] ✖ Updating iOS native dependencies with pod install - failed!
[capacitor] ✖ update ios - failed!
[capacitor] [error] Analyzing dependencies
[capacitor]         [!] CocoaPods could not find compatible versions for pod "GoogleMLKit/BarcodeScanning":
[capacitor]         In Podfile:
[capacitor]         CapacitorMlkitBarcodeScanning (from `../../node_modules/@capacitor-mlkit/barcode-scanning`) was resolved to 7.2.1, which depends on
[capacitor]         GoogleMLKit/BarcodeScanning (= 7.0.0)
[capacitor]         
[capacitor]         Specs satisfying the `GoogleMLKit/BarcodeScanning (= 7.0.0)` dependency were found, but they required a higher minimum deployment target.
[capacitor]         
[ERROR] An error occurred while running subprocess capacitor.
        
        capacitor sync ios exited with exit code 1.
        
        Re-running this command with the --verbose flag may provide more
        information.

--- a/.changeset/wet-ducks-pump.md
+++ b/.changeset/wet-ducks-pump.md
@@ -1,0 +1,5 @@
+---
+'@capacitor-mlkit/barcode-scanning': patch
+---
+
+fix(android): corner points were calculated somewhat inaccurately

--- a/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/BarcodeScanner.java
+++ b/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/BarcodeScanner.java
@@ -24,6 +24,8 @@ import androidx.camera.core.CameraSelector;
 import androidx.camera.core.ImageAnalysis;
 import androidx.camera.core.ImageProxy;
 import androidx.camera.core.Preview;
+import androidx.camera.core.resolutionselector.ResolutionSelector;
+import androidx.camera.core.resolutionselector.ResolutionStrategy;
 import androidx.camera.lifecycle.ProcessCameraProvider;
 import androidx.camera.view.PreviewView;
 import androidx.core.content.ContextCompat;
@@ -108,10 +110,15 @@ public class BarcodeScanner implements ImageAnalysis.Analyzer {
         BarcodeScannerOptions options = buildBarcodeScannerOptions(scanSettings);
         barcodeScannerInstance = BarcodeScanning.getClient(options);
 
+        ResolutionSelector resolutionSelector = new ResolutionSelector.Builder()
+            .setResolutionStrategy(new ResolutionStrategy(scanSettings.resolution, ResolutionStrategy.FALLBACK_RULE_CLOSEST_LOWER))
+            .build();
+
         ImageAnalysis imageAnalysis = new ImageAnalysis.Builder()
             .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
-            .setTargetResolution(scanSettings.resolution)
+            .setResolutionSelector(resolutionSelector)
             .build();
+
         imageAnalysis.setAnalyzer(ContextCompat.getMainExecutor(plugin.getContext()), this);
 
         ListenableFuture<ProcessCameraProvider> cameraProviderFuture = ProcessCameraProvider.getInstance(plugin.getContext());

